Dart/src/com/jetbrains/lang/dart/publicApi/ThirdPartyUtil.java

```diff
@@ -0,0 +1,42 @@
+package com.jetbrains.lang.dart.publicApi;
+
+import com.google.dart.server.AnalysisServerListener;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import com.jetbrains.lang.dart.ide.actions.DartStyleAction;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * This class is an access point for third party plugins to integrated Dart Plugin features in downstream plugins.
+ */
+@SuppressWarnings("unused")
+public class ThirdPartyUtil {
```

Comment:
Do you feel strong about the need for such a class?
I consider everything not private within Dart plugin code to be the public API. If we try to gather everything that is already used by Flutter plugin in this class it will already become enormous and harder to use.